File: javascript/easy/reverseInteger.js
var reverse = function(x) {
    let n = Math.abs(x), res = 0;
    while(n > 0) {
        res = (res * 10) + (n % 10);
        n = parseInt(n/10);
    }
    return (res > Math.pow(2, 31)) ? 0 : ((x < 0) ? -res : res);
};

File: ruby/easy/reverse_integer.rb
# @param {Integer} x
# @return {Integer}
def reverse(x)
  n, res = x.abs, 0
  while(n > 0)
    res = (res * 10) + (n%10)
    n = (n/10).to_i
  end
  return res > ((2 ** 31) - 1) ? 0 : (x < 0) ? -res : res
end